### ISSUE TITLE
[BITV/A11Y] Currently keyboard accessible using keyboard shortcuts common in text editors, however there is a WAI-ARIA best practice

### ISSUE THREAD

**Issue by AndyScherzinger** [nextcloud/text#2735]
Moved over from https://github.com/nextcloud/server/issues/21443

Details see [wcagFixes.xlsx](https://github.com/nextcloud/server/files/4788866/wcagFixes.xlsx) as well as discussions in https://github.com/nextcloud/server/issues/21443#issuecomment-645389427

c&op from the content:

> 4 | Keyboard | Currently keyboard accessible using keyboard shortcuts common in text editors, however there is a WAI-ARIA best practice | None | 2.1.1 - Text Editor |   | Text Editor is not configurable due to it being another third party application
> -- | -- | -- | -- | -- | -- | --

Is the missing `role="menubar"` on formatting bar as well as the missing `role="menuitem"` on each of the items what you are referring to? That we should of course fix.

Ref @juliushaertl for the Text editor:
- https://www.w3.org/TR/wai-aria/#menubar
- https://www.w3.org/TR/wai-aria/#menuitem

## ToDo

- [x] role menuitem/menubar https://github.com/nextcloud/text/pull/2715
- [ ] Expose shortcuts through [aria-keyshortcuts](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-keyshortcuts) #2762

**Comment by AndyScherzinger** [comment]
@juliushaertl assigned you just as a "ping"

**Comment by juliushaertl** [comment]
role menubar and menuitem were already addressed in https://github.com/nextcloud/text/pull/2715

**Comment by juliushaertl** [comment]
> Currently keyboard accessible using keyboard shortcuts common in text editors, however there is a WAI-ARIA best practice

As far as I get that this is about adding keyboard shortcuts through the aria-key shortcuts attribute: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-keyshortcuts

About the other topics I'm not sure I understand the report:

> Text Editor is not configurable due to it being another third party application

What should be configurable here? Is there a way we can clarify that?

**Comment by jancborchardt** [comment]
Tbh I also don't understand their remark @juliushaertl quoted.

Normally standard keyboard shortcuts should be ok as long as they are not single letters which might interfere with typing. And we don't use any of the like in Text, instead having standard shortcuts like Ctrl+B for bold, so that seems fine?

**Comment by juliushaertl** [comment]
Yes, we have pretty standard shortcuts there all involving some combination of Ctrl / Ctrl+Shift 👍

**Comment by AndyScherzinger** [comment]
@juliushaertl so did you also do?: Expose shortcuts through [aria-keyshortcuts](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-keyshortcuts) - if no then I think this is the only remaining issue looking at the ARIA spec.

**Comment by juliushaertl** [comment]
Yes, added as a todo item to the first post.
